Added HowTo guide for creating a plugin with the MinecraftDev IJ plugin

Leguan16 opened the pull request:
As communicated in the discord I took the initiative to add a tutorial for creating a project with help of the Minecraft Development IntelliJ plugin. 

I did not yet take a look at typos or grammar and just want to gather some feedback for the overall structure. 
I also changed the current `project-setup` markdown file to `manual-project-setup`. 
The `new_project_filled.png` currently is not in use. This might or might not be added later. 
For the list to explain the fields when creating a new project I am not sure if we should use a table instead. 

Feedback is appreciated!

powercasgamer commented on docs/paper/dev/getting-started/manual-project-setup.md:
```diff
@@ -1,9 +1,9 @@
 ---
-slug: /dev/project-setup
+slug: /dev/manual-project-setup
```
redirect might be required here?

Leguan16 replied:
from to? i cannot follow.

olijeffers0n replied:
yes, you cannot just change slugs. You need to set a redirect in the vercel file

powercasgamer commented on docs/paper/dev/getting-started/assisted-project-setup.md:
```diff
@@ -0,0 +1,68 @@
+* Optional Settings: Here you can define things like authors, website, description, etc. These are optional and not required for the plugin to work.
+* Build System: The build system you want to use. Paper recommends using Gradle but you can use Maven if you prefer.
+* Paper manifest: Whether you want to use the new Paper Plugins or not. For now this is not recommended as it is still in development.
+* Group ID: The group ID of your project. This is used for Maven and Gradle. This is usually your domain name in reverse. If you don't know what you should put here, you can use something like `com.<yourname>`
```
You should use a domain you own, so something like `io.github.<username>.<project>` or `me.<username>` is also commonly used

Leguan16 replied:
but you most likely don't own me.<name> and not everybody has a GitHub account. I don't think this is to change.

powercasgamer commented on docs/paper/dev/getting-started/assisted-project-setup.md:
```diff
@@ -0,0 +1,68 @@
+
+:::note
+
+If you want to streamline the process of testing a plugin, you can use the [Run-Paper](https://github.com/jpenilla/run-paper) Gradle task.
```
```suggestion
If you want to streamline the process of testing a plugin, you can use the [Run-Paper]https://github.com/jpenilla/run-task) Gradle plugin.
```

olijeffers0n commented on docs/paper/dev/getting-started/assisted-project-setup.md:
```diff
@@ -0,0 +1,68 @@
+---
+slug: /dev/assisted-project-setup
+description: Step-by-step instructions on how to setup a plugin with the MinecraftDev plugin.
```
Don’t like this slug

olijeffers0n commented on docs/paper/dev/getting-started/assisted-project-setup.md:
```diff
@@ -0,0 +1,68 @@
+
+This tutorial only works with IntelliJ IDEA. If you are using another IDE, please follow the [manual project setup guide](./manual-project-setup).
+
+:::
```
new line please

olijeffers0n commented on docs/paper/dev/getting-started/manual-project-setup.md:
```diff
@@ -1,9 +1,9 @@
 ---
 
-# Paper Project Setup
+# Manual Paper Project Setup
```
Bleh, don’t like manual. Just leave it as project setup and then have the mcdev example

Leguan16 replied:
alright

Leguan16 commented:
Alright,
I changed a few things:

- I moved `project-setup` and `mcdev-project-setup` to a separate subsection `project-setup`
- I changed the list to a table
- I reverted the slug

Now, with those changes I don't think we should leave the current project setup tutorial named "Paper Project Setup" since it should be obvious that we try to create a paper project. And it would look better if its just named "Project Setup" and the other one is called "McDev Project Setup". Maybe even just rename the title to "Manual Project Setup" and leave the slug how it is?

olijeffers0n commented:
Yeah so this is ready for content reviews - i’m happy with the addition of a new folder but you need to add a redirect for the slug change :)

Leguan16 commented:
> Yeah so this is ready for content reviews - i’m happy with the addition of a new folder but you need to add a redirect for the slug change :)

Dang it i forgot it again xd

Leguan16 commented:
I think i changed it?

olijeffers0n commented on docs/paper/dev/getting-started/project-setup/mcdev-project-setup.md:
```diff
@@ -0,0 +1,72 @@
+---
+slug: /dev/getting-started/project-setup/mcdev-project-setup
+description: Step-by-step instructions on how to setup a plugin with the MinecraftDev plugin.
```
I think clarifying the two “plugin”s is needed

Leguan16 replied:
Something like this?
```suggestion
description: Step-by-step instructions on how to setup a plugin with the MinecraftDev IntelliJ plugin.
```

olijeffers0n replied:
maybe change the first plugin to project and keep that Intelljj

Leguan16 replied:
Ok I changed it. Will push when I have more feedback.

olijeffers0n reviewed (changes requested):
Just one thing really, redirect lgtm

Leguan16 commented:
I moved the mcdev stuff over to misc and added a section for velocity plugins.
I also reverted the manual project setup back to its previous state.

olijeffers0n commented:
My apologies, that wasn't quite what I meant by misc. I meant `Paper/Dev/Miscellaneous` - Sorry

powercasgamer commented on docs/misc/mcdev-project-setup.md:
```diff
@@ -0,0 +1,101 @@
+| **Optional Settings** | Here you can define things like authors, website, description, etc. These are optional and not required for the plugin to work.                                                                                                                                           |
+| **Build System**      | The build system you want to use. Paper recommends using Gradle but you can use Maven if you prefer.                                                                                                                                                                      |
+| **Paper Manifest**    | Whether you want to use the new Paper plugins or not. For now this is not recommended as it is still in development.                                                                                                                                                      |
+| **Group ID**          | The group ID of your project. This is used for Maven and Gradle. This is usually your domain name in reverse. If you don't know what you should put here, you can use something like `io.github.<yourname>` or if you don't have GitHub you can just use `me.<yourname>`. |
```
```suggestion
| **Group ID**          | The group ID of your project. This is used for Maven and Gradle. This is usually your domain name in reverse. If you don't know what you should put here, you can use something like `io.github.<yourname>` or if you don't have GitHub you can use `me.<yourname>`. |
```

olijeffers0n commented:
Just needs a merge from main please 🙏
